kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -8,334 +9,295 @@ import scala.annotation.tailrec
+          * @return
+          *   The taken element
+          */
+        def take(using Frame): A < (Abort[Closed] & Async) =
+            IO.Unsafe {
+                self.poll().fold(Abort.error) {
+                    case Present(value) => value
+                    case Absent         => self.takeFiber().safe.get
+                }
+            }
 
-object Channel:
+        /** Creates a fiber that puts an element into the channel.
+          *
+          * @param value
+          *   The element to put
+          * @return
+          *   A fiber that completes when the element is put into the channel
+          */
+        def putFiber(value: A)(using Frame): Fiber[Closed, Unit] < IO = IO.Unsafe(self.putFiber(value).safe)
+
+        /** Creates a fiber that takes an element from the channel.
+          *
+          * @return
+          *   A fiber that completes with the taken element
+          */
+        def takeFiber(using Frame): Fiber[Closed, A] < IO = IO.Unsafe(self.takeFiber().safe)
+
+        /** Drains all elements from the channel.
+          *
+          * @return
+          *   A sequence containing all elements that were in the channel
+          */
+        def drain(using Frame): Seq[A] < (Abort[Closed] & IO) = IO.Unsafe(Abort.get(self.drain()))
+
+        /** Closes the channel.
+          *
+          * @return
+          *   A sequence of remaining elements
+          */
+        def close(using Frame): Maybe[Seq[A]] < IO = IO.Unsafe(self.close())
+
+        /** Checks if the channel is closed.
+          *
+          * @return
+          *   true if the channel is closed, false otherwise
+          */
+        def closed(using Frame): Boolean < IO = IO.Unsafe(self.closed())
+
+        /** Checks if the channel is empty.
+          *
+          * @return
+          *   true if the channel is empty, false otherwise
+          */
+        def empty(using Frame): Boolean < (Abort[Closed] & IO) = IO.Unsafe(Abort.get(self.empty()))
+
+        /** Checks if the channel is full.
+          *
+          * @return
+          *   true if the channel is full, false otherwise
+          */
+        def full(using Frame): Boolean < (Abort[Closed] & IO) = IO.Unsafe(Abort.get(self.full()))
+
+        def unsafe: Unsafe[A] = self
+    end extension
 
     /** Initializes a new Channel.
       *
       * @param capacity
-      *   The capacity of the channel
+      *   The capacity of the channel. Note that this will be rounded up to the next power of two.
       * @param access
       *   The access mode for the channel (default is MPMC)
-      * @param initFrame
-      *   The initial frame for the channel
       * @tparam A
       *   The type of elements in the channel
       * @return
       *   A new Channel instance
+      *
+      * @note
+      *   The actual capacity will be rounded up to the next power of two.
+      * @warning
+      *   The actual capacity may be larger than the specified capacity due to rounding.
```

Comment:
One of the two bugs I identified is due to JCTools rounding up the size of queues to the next power of two. I've decided to keep rounding for `Channel` and `Queue` and document the behavior. For `Meter`, the rounding is a major issue since, if the user requests 100 req/s, it'd become 128 req/s. I had to reimplement `Meter` with a different approach to fix it. More details below.